### PR TITLE
feat(engine) Add the auto-discover of the index.py files

### LINKED ISSUE #229: feature: autodiscover index config
Discover and load all modules (`index.py` for example) automatically. The module is responsible of defining the AlgoliaIndex and calling the `register(..., ...)` function.

It's a common pattern. For beginner, it will look like Django Admin. Super easy!

To discover module, it simple as that:
https://github.com/dfunckt/django-rules/blob/8765833c4474e7b6890627a58034966abaa2f403/rules/apps.py#L8-L11

(just got the idea, 2 years after 😂)

### PR DESCRIPTION
**Pull request by clemfromspace**
## TL;DR

- The `index.py` files are now auto-discovered when they exists
- A new `@register` decorator is now available

| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no     
| Related Issue     | fix #229
| Need Doc update   | yes

### Before
```python
# apps.py
from django.apps import AppConfig
import algoliasearch_django as algoliasearch

class YourAppConfig(AppConfig):
    name = 'your_app'

    def ready(self):
        YourModel = self.get_model('your_model')
        algoliasearch.register(YourModel)
```
### After
```python
# index.py
from algoliasearch_django import register
import algoliasearch_django as algoliasearch, AlgoliaIndex

from .models import YourModel

algoliasearch.register(YourModel)  # with the method

@register(YourModel)
class YourModelIndex(AlgoliaIndex): # with the decorator
    pass
```

## What was changed

The `index.py` files contained in the application modules are now "auto discovered" during the `AppReady` Django state.
Additionally, a `@register` decorator was added, allowing the user to directly register a `*Index` class with its pendant django `Model` class (See above for an example).

## Why it was changed

Theses changes means that the users don't need to create a `AppConfig` anymore, when they don't have one already. 
As stated in #229, it gives a more "Django feeling" since this method is used by the [`django.contrib.admin` package](https://docs.djangoproject.com/en/2.0/ref/contrib/admin/#modeladmin-objects).

### CODE REVIEW

**Comment by PLNech** [comment]
Don't forget to update the readme before merging :upside_down_face:

**Comment by clemfromspace** [comment]
@PLNech I updated the readme here: https://github.com/algolia/algoliasearch-django/pull/256/files#diff-04c6e90faac2675aa89e2176d2eec7d8
But I guess I should to a PR on the doc repo, right?

**Comment by PLNech** [comment]
@clemfromspace indeed you should do it on the docs repo :upside_down_face:

**Comment by clemfromspace** [comment]
@PLNech The doc is now up-to-date, I let you merge :)
